lib/src/results.dart

```diff
@@ -41,17 +42,14 @@ class RealmResults<T extends RealmObjectBase> extends collection.IterableBase<T>
 
   /// Returns the element of type `T` at the specified [index].
   T operator [](int index) {
-    final handle = realmCore.getObjectAt(this, index);
-    return realm.createObject(T, handle, _metadata!) as T;
-  }
-
-  /// Returns a new [RealmResults] filtered according to the provided query.
-  ///
-  /// The Realm Dart and Realm Flutter SDKs supports querying based on a language inspired by [NSPredicate](https://academy.realm.io/posts/nspredicate-cheatsheet/)
-  /// and [Predicate Programming Guide.](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/Predicates/AdditionalChapters/Introduction.html#//apple_ref/doc/uid/TP40001789)
-  RealmResults<T> query(String query, [List<Object> args = const []]) {
-    final handle = realmCore.queryResults(this, query, args);
-    return RealmResultsInternal.create<T>(handle, realm, _metadata);
+    final meta = _metadata;
```

Comment:
Why not check if T is RealmObjectBase which is the real marker if this can create objects or it is a results of primitives.

Reply:
We can check either one, but we should assert on the other - i.e. if we get a `RealmObjectBase` with no meta or meta with a non-RealmObjectBase generic, that would be a bug in the SDK.

Reply:
I agree we can assert the metadata, but I think we should  only check the type is `RealmObjectBase` since this is the real marker.

Reply:
I have reworked this to check on type instead. Assert on `_meta != null` is implicit in:
```dart
      final handle = realmCore.resultsGetObjectAt(this, index);
      final accessor = RealmCoreAccessor(metadata, realm.isInMigration);
```
I expect the first line to fail, and otherwise `metadata => _metadata!` will.